src/FlyCube/CommandList/CommandList.h

```diff
@@ -104,5 +104,7 @@ class CommandList : public QueryInterface {
                                   uint32_t query_count,
                                   const std::shared_ptr<Resource>& dst_buffer,
                                   uint64_t dst_offset) = 0;
+    virtual void SetGraphicsConstant(uint32_t root_parameter_index, uint32_t value, uint32_t byte_offset) = 0;
```

Comment:
It is better to use PushConstants, to match it with `[[vk::push_constant]]` in hlsl.
`void PushConstants(ShaderType shader_type, uint32_t dst_offset, const void* data, uint32_t size)`

In Vulkan we can have at most 1 push_constant block in each shader, I suggest to add the same limitation in FlyCube. In this case `ShaderType` + active `Pipeline` will probably be enough to find root_parameter_index in DirectX12 implementation.

In my opinion `PushConstants` is enough to for all cases, but I don't mind having `PushConstant` as well.
`void PushConstant(ShaderType shader_type, uint32_t dst_offset, uint32_t value)`